lib/guardian.rb

```diff
@@ -529,9 +529,18 @@ def can_see_site_contact_details?
   end
 
   def auth_token
-    if cookie = request&.cookies[Auth::DefaultCurrentUserProvider::TOKEN_COOKIE]
-      UserAuthToken.hash_token(cookie)
-    end
+    return if !request
+
+    cookie_name = Auth::DefaultCurrentUserProvider::TOKEN_COOKIE
+    cookie = request.cookie_jar[cookie_name].presence
+    return if !cookie
+
+    # v0 of our auth cookie
+    return UserAuthToken.hash_token(cookie) if cookie.size == 32
+
+    cookie = request.cookie_jar.encrypted[cookie_name].presence
+    return if !cookie
```

Comment:
Hmm do we need this when we've already checked the presence of the cookie above.

Reply:
yeah we still need it because the cookie might decrypt correctly in which case we'll get a nil. However, we can probably remove `.presence` from line 542.